### PR TITLE
#94: removed name from upload form

### PR DESCRIPTION
**Pull request by ggajos**
for  #94

### CODE REVIEW

**Comment by karato** [comment]
Thanks, let me find someone who can review this pull request

**Comment by karato** [comment]
@alevohin review it

File: src/main/java/com/nerodesk/takes/doc/TsDoc.java [hunk 1]
```diff
@@ -70,17 +71,26 @@ public TsDoc(final Base bse) {
...
+    //  from `Takes` framework or implement in this project class that is going
+    //  to do this in clean way.
     @Override
     public Take route(final Request req) throws IOException {
         final String file;
         final Href href = new RqHref(req).href();
-        final Iterator<String> param = href.param("file").iterator();
+        final String key = "file";
+        final Iterator<String> param = href.param(key).iterator();
         if (param.hasNext()) {
```
Comment (alevohin) [inline src/main/java/com/nerodesk/takes/doc/TsDoc.java:89]: Do we really need this `if/else`? `filename` always is passed as part of multipart request?

**Comment by alevohin** [comment]
@ggajos could you add unit test for `TsDoc` for your changes?

**Comment by alevohin** [comment]
@ggajos Travis build failed with
`TsAppTest.uploadsFileContent:197 » FileNotFound /tmp/1427359831915-2/urn/test/...`
please see full log for details

**Comment by alevohin** [comment]
@ggajos thank you! Please see comments above.

File: src/main/java/com/nerodesk/takes/doc/TsDoc.java (hunk 1 above)
Comment (ggajos) [inline src/main/java/com/nerodesk/takes/doc/TsDoc.java:89]: I think we do. This method handles all types of request upload/remove; sometimes is part of upload; sometimes is as separate input `name`.

**Comment by ggajos** [comment]
@alevohin thanks, I had lot's of merge conflicts here, I had to overlook something, added another PDD for tests (too time consuming to do it in single approach). Anyway thanks for feedback, please have a look if everything is fine now

File: src/main/java/com/nerodesk/takes/doc/TsDoc.java (hunk 1 above)
Comment (alevohin) [inline src/main/java/com/nerodesk/takes/doc/TsDoc.java:89]: Totally agree, thank you!

File: src/main/java/com/nerodesk/takes/doc/TsDoc.java [hunk 2]
```diff
@@ -72,17 +73,29 @@ public TsDoc(final Base bse) {
         this.base = bse;
     }
 
+    // @todo #94:30min I tried to get filename from file meta data and it
+    //  doesn't look clean but works for now. We have to either wait for impl.
+    //  from `Takes` framework or implement in this project class that is going
+    //  to do this in clean way.
+    // @todo #94:30min This method is way to complicated and should be splitted
+    //  to smaller chunks. It's very hard to test what is going on here. After
+    //  splitting it can be covered by unit tests.
```
Comment (alevohin) [inline src/main/java/com/nerodesk/takes/doc/TsDoc.java]: l suggest replace `it can be covered` to `it should be covered`

**Comment by alevohin** [comment]
@ggajos great! Please see one minor comment.

**Comment by ggajos** [comment]
@alevohin thanks for feedback, please have a look if everything is fine now

**Comment by alevohin** [comment]
@rultor merge @ggajos thank you!

**Comment by yegor256** [comment]
@rultor merge

**Comment by karato** [comment]
@alevohin Thanks for your contribution, **20 mins** was added to your account, payment ID is `54269262`, 22 hours and 40 mins spent total; extra minutes for review comments (c=5); +20 added to your rating, at the moment it is: [+2091](http://www.netbout.com/b/35105?open=rating)

**Comment by karato** [comment]
@rultor deploy now pls

**Pull request merged**